scaleway/resource_instance_image_test.go

```diff
@@ -0,0 +1,393 @@
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testAccCheckScalewayInstanceImageDestroy(tt),
+			testAccCheckScalewayInstanceSnapshotDestroy(tt),
+			testAccCheckScalewayInstanceVolumeDestroy(tt),
+			testAccCheckScalewayInstanceServerDestroy(tt),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_instance_server" "main" {
+						image = "ubuntu_focal"
+						type = "DEV1-S"
+						state = "stopped"
+					}
+
+					resource "scaleway_instance_volume" "main" {
+						type       = "l_ssd"
+						size_in_gb = 20
+						zone = "nl-ams-1"
+					}
+
+					resource "scaleway_instance_snapshot" "main" {
+						volume_id = scaleway_instance_server.main.root_volume.0.volume_id
+					}
+
+					resource "scaleway_instance_image" "main" {
+						root_volume_id = scaleway_instance_snapshot.main.id
+						architecture = "x86_64"
+					}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayInstanceImageExists(tt, "scaleway_instance_image.main"),
+					resource.TestCheckResourceAttrSet("scaleway_instance_image.main", "name"),
+					resource.TestCheckResourceAttr("scaleway_instance_image.main", "architecture", "x86_64"),
+					resource.TestCheckResourceAttr("scaleway_instance_image.main", "public", "false"),
+					resource.TestCheckResourceAttrPair("scaleway_instance_image.main", "root_volume_id", "scaleway_instance_snapshot.main", "id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScalewayInstanceImage_SeveralVolumes(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testAccCheckScalewayInstanceImageDestroy(tt),
+			testAccCheckScalewayInstanceSnapshotDestroy(tt),
+			testAccCheckScalewayInstanceVolumeDestroy(tt),
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+					resource "scaleway_instance_volume" "vol00" {
+						type       = "b_ssd"
+						size_in_gb = 20
+					}
+					resource "scaleway_instance_snapshot" "snap00" {
+						volume_id = scaleway_instance_volume.vol00.id
+					}
+
+					resource "scaleway_instance_volume" "vol01" {
+						type       = "b_ssd"
+						size_in_gb = 20
+					}
+					resource "scaleway_instance_snapshot" "snap01" {
+						volume_id = scaleway_instance_volume.vol01.id
+					}
+
+					resource "scaleway_instance_volume" "vol02" {
+						type       = "b_ssd"
+						size_in_gb = 20
+					}
+					resource "scaleway_instance_snapshot" "snap02" {
+						volume_id = scaleway_instance_volume.vol02.id
+					}
+
+					resource "scaleway_instance_image" "main" {
+						root_volume_id = 	scaleway_instance_snapshot.snap00.id
+						architecture = 		"x86_64"
```

Comment:
Is the architecture always assumed to be x86_64 ? If that is the case, let's just remove it altogether and not specify it explicitly so that it is handled in the backend.